tools/preflight/preflight.go

```diff
@@ -0,0 +1,620 @@
+	}
+
+	//  Check storage snapshot class
+	o.Logger.Infoln("Checking if a StorageClass and VolumeSnapshotClass are present")
+	err = o.checkStorageSnapshotClass()
+	if err != nil {
+		o.Logger.Errorln(fmt.Sprintf("%s Preflight check for SnapshotClass failed :: %s", cross, err.Error()))
+		storageSnapshotSuccess = false
+		preflightStatus = false
+	} else {
+		o.Logger.Infoln(fmt.Sprintf("%s Preflight check for SnapshotClass is successful", check))
+	}
+
+	//  Check CSI installation
+	o.Logger.Infoln("Checking if CSI APIs are installed in the cluster")
+	err = o.checkCSI()
+	if err != nil {
+		o.Logger.Errorln(fmt.Sprintf("Preflight check for CSI failed :: %s", err.Error()))
+		preflightStatus = false
+	} else {
+		o.Logger.Infoln(fmt.Sprintf("%s Preflight check for CSI is successful", check))
+	}
+
+	//  Check DNS resolution
+	o.Logger.Infoln("Checking if DNS resolution is working in k8s cluster")
+	err = o.checkDNSResolution()
+	if err != nil {
+		o.Logger.Errorln(fmt.Sprintf("%s Preflight check for DNS resolution failed :: %s", cross, err.Error()))
+		preflightStatus = false
+	} else {
+		o.Logger.Infoln(fmt.Sprintf("%s Preflight check for DNS resolution is successful", check))
+	}
+
+	//  Check volume snapshot and restore
+	if storageSnapshotSuccess {
+		o.Logger.Infoln("Checking if volume snapshot and restore is enabled in cluster")
+		err = o.checkVolumeSnapshot()
+		if err != nil {
+			o.Logger.Errorln(fmt.Sprintf("%s Preflight check for volume snapshot and restore failed :: %s", cross, err.Error()))
+			preflightStatus = false
+		} else {
+			o.Logger.Infoln(fmt.Sprintf("%s Preflight check for volume snapshot and restore is successful", check))
+		}
+	}
+
+	co := &CleanupOptions{
+		Ctx:        o.Context,
+		Kubeconfig: o.Kubeconfig,
+		Namespace:  o.Namespace,
+		Logger:     o.Logger,
+	}
+	if !preflightStatus {
+		o.Logger.Warnln("Some preflight checks failed")
+		if o.PerformCleanupOnFail {
+			if err = co.CleanupByUID(resNameSuffix); err != nil {
+				o.Logger.Errorln(fmt.Sprintf("%s Failed to cleanup preflight resources :: %s", cross, err.Error()))
+			}
+		}
+	} else {
+		o.Logger.Infoln("All preflight checks succeeded!")
+		if err = co.CleanupByUID(resNameSuffix); err != nil {
+			o.Logger.Errorln(fmt.Sprintf("%s Failed to cleanup preflight resources :: %s", cross, err.Error()))
+		}
+	}
+}
+
+// checkKubectl checks whether kubectl utility is installed.
+func (o *Options) checkKubectl() error {
+	path, err := goexec.LookPath("kubectl")
+	if err != nil {
+		return fmt.Errorf("error finding 'kubectl' binary in $PATH of the system :: %s", err.Error())
+	}
+	o.Logger.Infoln(fmt.Sprintf("kubectl found at path - %s", path))
+
+	return nil
+}
+
+// checkClusterAccess Checks whether access to kubectl utility is present on the client machine.
+func (o *Options) checkClusterAccess() error {
+	cmdOut, err := shell.RunCmd(fmt.Sprintf("kubectl cluster-info --kubeconfig=%s", o.Kubeconfig))
```

Comment:
we'd concluded to use GET ns by go-client instead of bash cmd... why not changed here?